clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -720,14 +729,26 @@ public FetchResponseMetricAggregator(FetchManagerMetrics sensors,
          * and number of records parsed. After all partitions have reported, we write the metric.
          */
         public void record(TopicPartition partition, int bytes, int records) {
-            unrecordedPartitions.remove(partition);
-            totalBytes += bytes;
-            totalRecords += records;
+            this.unrecordedPartitions.remove(partition);
+            this.fetchMetrics.increment(bytes, records);
 
-            if (unrecordedPartitions.isEmpty()) {
+            // collect and aggregate per-topic metrics
+            String topic = partition.topic();
+            if (this.topicFetchMetrics.get(topic) == null)
+                this.topicFetchMetrics.put(topic, new FetchMetrics());
+            FetchMetrics topicFetchMetric = this.topicFetchMetrics.get(topic);
```

Comment:
Haha, to clarify, I was trying to suggest moving this line above null check. Something like this:
```java
FetchMetrics topicFetchMetric = this.topicFetchMetrics.get(topic);
if (topicFetchMetric == null) {
  topicFetchMetric = new FetchMetrics();
  this.topicFetchMetrics.put(topic, topicFetchMetric);
}
```